### PR TITLE
#166151266 Export/Import Workouts

### PR DESCRIPTION
**Pull request by wangonya**
### What does this PR do?
Allows user to export workout to json file and import from json.

### Description of Task to be completed?
- [x] Add export workout as json
- [x] Add import workout from json

#### Exporting a workout
* Start server by running `python manage.py runserver`
* Navigate to `http://127.0.0.1:8000/en/workout/overview` and click on a workout. Create one if none is available.
* Add exercises to workout
* Click on `Options` -> `Export workout data` -> `JSON`

#### Importing a workout
* Navigate to `http://127.0.0.1:8000/en/workout/overview` and click on `Add workout`
* Click on `Import from JSON`
* Choose the file you exported earlier

### What are the relevant pivotal tracker stories?
[#166151266](https://www.pivotaltracker.com/story/show/166151266) 

### Screenshots
Screenshot 2019-06-20 at 09 46 55
Screenshot 2019-06-20 at 09 46 39
Screenshot 2019-06-19 at 06 47 16

### CODE REVIEW

File: wger/manager/static/img/json.svg [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			s0.41,0.111,0.629,0.157s0.424,0.068,0.615,0.068c0.483,0,0.868-0.094,1.155-0.28s0.439-0.504,0.458-0.95v-7.711H17.021z"/>
+		<path style="fill:#FFFFFF;" d="M25.184,50.238c0,0.364-0.075,0.718-0.226,1.06s-0.362,0.643-0.636,0.902s-0.61,0.467-1.012,0.622
+			s-0.856,0.232-1.367,0.232c-0.219,0-0.444-0.012-0.677-0.034s-0.467-0.062-0.704-0.116c-0.237-0.055-0.463-0.13-0.677-0.226
+			s-0.398-0.212-0.554-0.349l0.287-1.176c0.128,0.073,0.289,0.144,0.485,0.212s0.398,0.132,0.608,0.191s0.419,0.107,0.629,0.144
+			s0.405,0.055,0.588,0.055c0.556,0,0.982-0.13,1.278-0.39s0.444-0.645,0.444-1.155c0-0.31-0.104-0.574-0.314-0.793
+			s-0.472-0.417-0.786-0.595s-0.654-0.355-1.019-0.533s-0.706-0.388-1.025-0.629s-0.583-0.526-0.793-0.854s-0.314-0.738-0.314-1.23
+			c0-0.446,0.082-0.843,0.246-1.189s0.385-0.641,0.663-0.882s0.602-0.426,0.971-0.554s0.759-0.191,1.169-0.191
+			c0.419,0,0.843,0.039,1.271,0.116s0.774,0.203,1.039,0.376c-0.055,0.118-0.118,0.248-0.191,0.39s-0.142,0.273-0.205,0.396
+			s-0.118,0.226-0.164,0.308s-0.073,0.128-0.082,0.137c-0.055-0.027-0.116-0.063-0.185-0.109s-0.166-0.091-0.294-0.137
```
Comment (wekesa931) [inline wger/manager/static/img/json.svg:23]: Is this file autogenerated or did you make it yourself?

File: wger/manager/templates/workout/view.html [hunk 2]
```diff
@@ -219,12 +262,22 @@ <h4 class="modal-title">{% trans "Download as PDF" %}</h4>
         {% trans "Options" %}
     </button>
     <ul class="dropdown-menu">
+    {% if workout.canonical_representation.day_list %}
         <li>
-            <a data-toggle="modal" data-target="#download-pdf-popup">
+            <a data-toggle="modal" data-target="#export-workout-popup">
```
Comment (wekesa931) [inline wger/manager/templates/workout/view.html:267]: Why did you change the target-data here? don't you think this might affect other areas of the code?

File: wger/manager/templates/workout/view.html [hunk 3]
```diff
@@ -144,6 +144,49 @@ <h4 class="modal-title">{% trans "Export calendar file" %}</h4>
...
+            </div>
+            <div class="modal-body">
+                <form class="form-horizontal">
+                    <div class="form-group">
+                        <div>
+                            <label for="id_format" class="col-md-3 control-label">
+                                {% trans "Format: " %}
+                            </label>
+                        </div>
+                        <div class="col-md-9 row" style="padding-left: 35px;">
```
Comment (jmusila) [inline wger/manager/templates/workout/view.html]: Maybe you should have added styling in CSS file

File: wger/manager/static/img/json.svg (hunk 1 above)
Comment (wangonya) [reply wger/manager/static/img/json.svg:23]: This is an `svg` image. I downloaded it and added it here manually.

File: wger/manager/tests/test_workout.py [hunk 4]
```diff
@@ -202,3 +207,74 @@ class WorkoutApiTestCase(api_base_test.ApiBaseResourceTestCase):
     private_resource = True
     special_endpoints = ('canonical_representation',)
     data = {'comment': 'A new comment'}
+
+
+class WorkoutExportImportTestCase(WorkoutManagerTestCase):
+    '''
+    Tests exporting importing json
```
Comment (jmusila) [inline wger/manager/tests/test_workout.py:214]: Good job on the testing 👍

File: wger/manager/templates/workout/view.html (hunk 2 above)
Comment (wangonya) [reply wger/manager/templates/workout/view.html:267]: It does not affect other areas of the code. Before, there was a `download as pdf` link here. I found it repetitive to add a `download as json` below it since they do almost the same thing. So I removed `download-pdf-popup` and added a general `export-workout-popup`, then gave the option to download as `pdf` or `json` in the popup as can be seen in the first screenshot of the description.

Have you tried testing the feature locally? That might help you understand why the change here was necessary.

**Comment by wekesa931** [comment]
You have done a great job! The feature works well, the syntax is okay and the fact that you have finished the work in record time shows the commitment you have channeled to this work. Well done. 👏🏻

**Review by jmusila — COMMENTED** [review]
Tested locally, works as stated. Great job on the pdf and json svg's.

File: wger/manager/templates/workout/view.html (hunk 3 above)
Comment (wangonya) [reply wger/manager/templates/workout/view.html]: Noted. I've done this.

**Comment by ceexon** [comment]
I've tested and its working

**Review by ceexon — APPROVED** [review]

File: wger/manager/models.py [hunk 5]
```diff
@@ -69,6 +69,7 @@ class Meta:
                     "example 'Focus on back' or 'Week 1 of program xy'."))
     user = models.ForeignKey(User,
                              verbose_name=_('User'), on_delete=models.CASCADE)
+    imported = models.BooleanField(blank=True, default=False)
```
Comment (willeswa) [inline wger/manager/models.py]: Since you have set the default to False, is it necessary to set blank to True?

Comment (wangonya) [reply wger/manager/models.py]: Nice catch. I don't think it's necessary. I'll remove it.
